baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/IllegalSafeLoggingLambdaTest.java

```diff
@@ -86,6 +86,90 @@ void testNestedAnonymousInLambdaUnsafeExternalData() {
                 .doTest();
     }
 
+    //    @Test
+    //    public void testUnsafeStatementLambdaSupplier() {
+    //        helper().addSourceLines(
+    //                        "Test.java",
+    //                        "import com.palantir.logsafe.*;",
+    //                        "import java.util.function.*;",
+    //                        "class Test {",
+    //                        "  static void f(@Unsafe Object value) {",
+    //                        "    // BUG: Diagnostic contains: Dangerous argument value: arg is 'UNSAFE'",
+    //                        "    Supplier<@Safe Object> supplier = () -> {return value;};",
+    //                        "  }",
+    //                        "}")
+    //                .doTest();
+    //    }
```

Comment:
Will either remove or make it support statement lambdas, if I can figure it out and it's not too much extra code (otherwise I'll move to a later PR)